ehrapy/tools/feature_ranking/_rank_features_groups.py

```diff
@@ -320,8 +325,54 @@ def rank_features_groups(
          >>> ep.tl.rank_features_groups(adata, "service_unit")
          >>> ep.pl.rank_features_groups(adata)
     """
+    # if rank_obs_columns is indicated, layer must be None
+    if layer is not None and rank_obs_columns is not None:
+        raise ValueError("Only one of layer and rank_obs_columns can be specified.")
+
     adata = adata.copy() if copy else adata
 
+    if rank_obs_columns is not None:
+        # keep reference to original adata, needed if copy=False
+        adata_orig = adata
+        # copy adata to work on
+        adata = adata.copy()
```

Comment:
Such a copy can be very expensive. Is it possible to instead create an intermediate AnnData object with only the information that we need?

Reply:
Yes I think so - the latest commit only keeps the `.obs `field (and for conciseness one dummy column of zeros in `.X`)